404 and offline prototypes

AriannaChau opened the pull request:
## Overview

This PR adds two new pages: A 404 page and an offline page. Originally, these pages were going to look more similar to what we have on the live site, but along the way they’ve taken on a new layout specific to them.
I’m submitting this PR because the design is clear but there are some technical design things that should be ironed out like: 
- A simpler cloud footer.
- Larger heading option(s)
- Different robot art (I've opened up a separate issue for this task: https://github.com/cloudfour/cloudfour.com-patterns/issues/1555)

I've already created a new issue for the robot illustration, and plan on creating another one to iron out these details that weren't highlighted in the original issue.

## Screenshots

### 404 page: 
<img width="1589" alt="Screen Shot 2021-09-24 at 3 49 12 PM" src="https://user-images.githubusercontent.com/24928337/134747046-08653064-00ad-4fe3-8964-89f81e884b5f.png">

### Offline page: 
<img width="1767" alt="Screen Shot 2021-09-24 at 3 49 28 PM" src="https://user-images.githubusercontent.com/24928337/134747062-92832b29-e6f2-4501-a0fd-253da92626e8.png">


## Testing

1. [Deploy Link (404)](https://deploy-preview-1554--cloudfour-patterns.netlify.app/iframe.html?id=prototypes-error-pages--error-404&args=&viewMode=story)
2. [Deploy link (offline)](https://deploy-preview-1554--cloudfour-patterns.netlify.app/iframe.html?id=prototypes-error-pages--offline&viewMode=story)

---

- Fixes https://github.com/cloudfour/cloudfour.com-patterns/issues/1546

tylersticka reviewed (commented):
Overall, I dig the simplified layouts a lot!

Some feedback...

1. The spacing between the heading, paragraph and search field seems odd to me. Why is there so much more space above compared to below, for example?
2. The `line-height` of the headings feels like a lot. I would suggest sticking with the level -2 size of [our heading component for now](https://v-next--cloudfour-patterns.netlify.app/iframe.html?id=components-heading--example&viewMode=docs&utm_source=github&utm_campaign=bot_dp#levels). If we need to, we could also add additional negative step sizes for larger headings.
3. At small sizes, the copyright text can pass over the clouds and render the contrast inaccessible.

AriannaChau commented:
1. I didn't add any additional padding so I think it had to do with the line-height. I added a level to the heading and that seemed to fix it a bit.
2. I adjusted the footer.

tylersticka reviewed (commented):
The latest revision is an improvement. Additional feedback...

1. The heading sizes still feel not as fluid as our standard ones. Can we use level -2 without the custom sizing styles?
2. Some viewport sizes seem to squash the content to an uncomfortably narrow degree, see screenshot below.

<img width="1009" alt="Screen Shot 2021-09-27 at 1 28 38 PM" src="https://user-images.githubusercontent.com/69633/134980515-44d5e6fd-4da4-4f4b-a085-5db627af0bf6.png">
